src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/Infrastructure.kt

```diff
@@ -73,6 +80,29 @@ class Infrastructure<out T : VirtualUsers>(
     }
 }
 
+interface TargetingVirtualUserOptions {
+
+    fun target(
+        jira: URI
+    ): VirtualUserOptions
+}
+
+private class LegacyTargetingVirtualUserOptions(
+    private val behavior: VirtualUserBehavior
+) : TargetingVirtualUserOptions {
+
+    override fun target(
+        jira: URI
+    ): VirtualUserOptions = VirtualUserOptions(
+        target = VirtualUserTarget(
+            webApplication = jira,
+            userName = "admin",
+            password = "admin"
+        ),
+        behavior = behavior
```

Comment:
@sebapawlak, @wyrzyk you were right, the arbitrary behavior/target split was not the proper long-term solution.
We were trying to let aws-infra inject the Jira URI into VUO. It turns out we can ask for this directly 🤦‍♂️